engine/src/main/java/org/terasology/logic/console/commands/CoreCommands.java

```diff
@@ -516,14 +536,139 @@ public String spawnBlock(@Sender EntityRef sender, @CommandParam("blockName") St
         blockItem.send(new DropItemEvent(spawnPos));
         return "Spawned block.";
     }
+    @Command(shortDescription = "spawns many block together (value up to you prefer)", helpText = "you can select the particular block you want to spawn" +
+            "include the amount", runOnServer = true, requiredPermission = PermissionManager.CHEAT_PERMISSION)
+    public String bulkDropper(@Sender EntityRef sender, @CommandParam("blockName") String blockName, @CommandParam("value") int value) {
+
+        //This is a loop which gives the particular amount of block the player wants to spawn
+        for (int i = 0; i <= value; i++) {
+            ClientComponent clientComponent = sender.getComponent(ClientComponent.class);
+            LocationComponent characterLocation = clientComponent.character.getComponent(LocationComponent.class);
+
+            Vector3f spawnPos = characterLocation.getWorldPosition();
+            Vector3f offset = characterLocation.getWorldDirection();
+            offset.scale(3);
+            spawnPos.add(5,10,0);
+
+            BlockFamily block = blockManager.getBlockFamily(blockName);
+            if (block == null) {
+                return "";
+            }
+            BlockItemFactory blockItemFactory = new BlockItemFactory(entityManager);
```

Comment:
Even though the compiler will move some of it outside the for loop, I'd like to see you move it all outside. Only the last two lines should be in the loop.

Reply:
These should still be outside the loop. You only need two lines inside the loop